modal/sandbox.py

```diff
@@ -626,6 +636,20 @@ async def tunnels(self, timeout: int = 50) -> dict[int, Tunnel]:
 
         return self._tunnels
 
+    async def create_connect_token(
+        self, metadata: Optional[Union[str, dict[str, Any]]] = None
+    ) -> SandboxConnectCredentials:
+        """Create a token for making HTTP connections to the sandbox."""
+        if metadata is not None and isinstance(metadata, dict):
+            try:
+                metadata = json.dumps(metadata)
+            except BaseException as e:
```

Comment:
You very rarely want to catch `BaseException` in Python; control-related exceptions like `KeyboardInterrupt` derive from it. `except Exception as e:` is the recommended spelling when you want "any" exception that might be triggered by what you do in the `try` block.

Reply:
Glad I could get a useful contribution in under the wire 😅